modules/gateway/peers.go

```diff
@@ -58,6 +68,14 @@ func (s insufficientVersionError) Error() string {
 	return "unacceptable version: " + string(s)
 }
 
+// invalidVersionError indicates a peer's version is not a valid version number.
+type invalidVersionError string
+
+// Error implements the error interface for invalidVersionError.
+func (s invalidVersionError) Error() string {
+	return "invalid version: " + string(s)
+}
```

Comment:
It seems to me like this error is only being used to be returned and for testing. I think it's probably okay to remove this type and just call `errors.New()` for most of what you are returning.